components/jenkins/data.tf

```diff
@@ -4,10 +4,27 @@ data "azurerm_resource_group" "disks_resource_group" {
   name = "disks-${var.env}-rg"
 }
 
+data "azurerm_service_principal" "service_principle" {
```

Comment:
This should be `azuread_service_principal`
https://registry.terraform.io/providers/hashicorp/azuread/1.6.0/docs/data-sources/service_principal